internal/agent/device/lifecycle/manager.go

```diff
@@ -92,15 +99,71 @@ func (m *LifecycleManager) Initialize(ctx context.Context, status *v1alpha1.Devi
 }
 
 func (m *LifecycleManager) Sync(ctx context.Context, current, desired *v1alpha1.RenderedDeviceSpec) error {
-	// TODO: implement
+	// this controller currently does not implement a sync operation
 	return nil
 }
 
 func (m *LifecycleManager) AfterUpdate(ctx context.Context, current, desired *v1alpha1.RenderedDeviceSpec) error {
-	// TODO: implement
+	if current.Decommission == nil && desired.Decommission != nil {
+		errored := false
+		m.log.Warn("Detected decommissioning request from flightctl service")
+		m.updateWithStartedCondition(ctx)
+		m.log.Warn("Updated Condition to decommissioning started")
+
+		// TODO: add support for additional decommissioning target types
+
+		if !errored {
+			m.updateWithCompletedCondition(ctx)
```

Comment:
The status is set to starting and immediately set to completed without any action in-between.  So what is the purpose of setting it to "started" to begin with ?

Reply:
It is a reasonable question status updates are not free so we should make sure they are required. At a minimum document intent if it is future functionality.

Reply:
Yes, it is future functionality. Currently we only support "unenroll" which basically will just do `wipeAndReboot`. In the future other targets will add further handling steps between `Started` and `Completed`. But this is stubbing out the intent here. I can add a more detailed note

Reply:
The service side expects the `Started` Condition followed by the `Completed` Condition, so we can't skip these.